### PR TITLE
Mediasaving

### PR DESCRIPTION
**Pull request by Andreas-Kreouzos**
Started the implementation with AWS S3 to save and getting images. At first, we added the necessary dependencies in pom.xml and wrote dummy credentials fields into application.properties file. Then, we proceeded as below:
1. Wrote the S3Config client class to configure the AWS S3
2. Wrote Image Service class to provide the business logic on storing and getting the images
3. Wrote the Image Controller class to create the necessary endpoints for the aforementioned functions
4. Wrote unit tests for both of these classes
5. Created an abstract Generic Controller Class to reduce code duplication in all of our controllers
6. Removed @Getter and @Setter with @Data of Lombok

### CODE REVIEW

File: pom.xml [hunk 1]
```diff
@@ -93,6 +93,18 @@
             <version>1.5.3.Final</version>
         </dependency>
 
+        <dependency>
+            <groupId>com.amazonaws</groupId>
+            <artifactId>aws-java-sdk</artifactId>
+            <version>1.12.153</version>
```
Comment (amigoscode) [inline pom.xml]: use v2

File: pom.xml [hunk 2]
```diff
@@ -93,6 +93,18 @@
...
+        </dependency>
+
+        <dependency>
```
Comment (amigoscode) [inline pom.xml:102]: maybe not needed

File: src/main/java/com/project/trackfit/aws/ImageController.java [hunk 3]
```diff
@@ -0,0 +1,44 @@
...
+
+import java.io.IOException;
+import java.util.UUID;
+
+@RestController
+@AllArgsConstructor
+@RequestMapping("api/v1/images")
+public class ImageController {
+
+    private ImageService imageService;
```
Comment (amigoscode) [inline src/main/java/com/project/trackfit/aws/ImageController.java]:
```suggestion
    private final ImageService imageService;
```

File: src/main/java/com/project/trackfit/aws/ImageController.java [hunk 4]
```diff
@@ -0,0 +1,44 @@
...
+    private CustomerService customerService;
```
Comment (amigoscode) [inline src/main/java/com/project/trackfit/aws/ImageController.java]:
```suggestion
    private final CustomerService customerService;
```

File: src/main/java/com/project/trackfit/aws/ImageController.java [hunk 5]
```diff
@@ -0,0 +1,44 @@
...
+
+    @PostMapping("{customerId}/upload")
+    public ResponseEntity<Media> uploadImage(
+            @PathVariable("customerId") UUID customerId,
+            @RequestParam("image") MultipartFile image) throws IOException {
+        Customer customer = customerService.getCustomerById(customerId);
```
Comment (amigoscode) [inline src/main/java/com/project/trackfit/aws/ImageController.java]: move this logic to service class

**Review by ghost — DISMISSED** [review]

File: src/main/java/com/project/trackfit/aws/ImageController.java (hunk 3 above)
Comment (Andreas-Kreouzos) [reply src/main/java/com/project/trackfit/aws/ImageController.java]: Done

File: src/main/java/com/project/trackfit/aws/ImageController.java (hunk 4 above)
Comment (Andreas-Kreouzos) [reply src/main/java/com/project/trackfit/aws/ImageController.java]: Done

File: pom.xml (hunk 1 above)
Comment (Andreas-Kreouzos) [reply pom.xml]: Removed the above code snippet and added the following:
```
        <dependency>
            <groupId>software.amazon.awssdk</groupId>
            <artifactId>s3</artifactId>
            <version>2.20.26</version>
        </dependency>
```

File: pom.xml (hunk 2 above)
Comment (Andreas-Kreouzos) [reply pom.xml:102]: Added version 2 instead of 1

File: src/main/java/com/project/trackfit/aws/ImageController.java (hunk 5 above)
Comment (Andreas-Kreouzos) [reply src/main/java/com/project/trackfit/aws/ImageController.java]: Done

**Review by ghost — APPROVED** [review]

**Pull request merged**